### ISSUE TITLE
No early stopping implemented in Chapter 3 (yelp)

### ISSUE THREAD

**Issue by lisanka93** [joosthub/PyTorchNLPBook#34]
The early_stopping_best_val never gets updated so the loss is ALWAYS smaller and early stopping never happens
this line of code is missing in the if-else statement in update_train_stage function:

train_state['early_stopping_best_val'] = loss_t

**Comment by Rucious-Aladdin** [comment]
please modify the syntax inside the **update_train_state** function as follows and ensure that the minimum value is initialized.

def update_train_state(args, model, train_state):
"""
codes...
"""
&nbsp;&nbsp;&nbsp;&nbsp;if loss_t < train_state['early_stopping_best_val']:
                &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; torch.save(model.state_dict(), train_state['model_filename'])
                &nbsp;&nbsp;&nbsp;&nbsp; &nbsp;&nbsp;&nbsp;&nbsp;**train_state['early_stopping_best_val'] = loss_t**
